universal-login-example/src/components/Backup.js

```diff
@@ -1,14 +1,80 @@
 import React, { Component } from 'react';
 import BackupView from '../views/BackupView';
 import PropTypes from 'prop-types';
+import { toWords } from '../Daefen';
+import ethers from 'ethers';
+import DEFAULT_PAYMENT_OPTIONS from '../../config/defaultPaymentOptions';
 
 class Backup extends Component {
+  constructor(props) {
+    super(props);
+    this.identityService = this.props.services.identityService;
+    this.sdk = this.props.services.sdk;
+    this.state = {
+      backupCodes: [],
+      publicKeys: [],
+      isLoading: false
+    };
+  }
+
+  componentDidMount() {
+    this.generateBackupCodes();
+  }
+
+  async generateBackupCodes() {
```

Comment:
I would extract this one to a separate class, let's say `BackupService`, that would have one key method `generateBackupCodes()` that would return a pair of arrays (publicKeys, code). So that component will mount would look like following:
```js
componentDidMount() {
  const [backupCodes, publicKeys] = this.props.backupService.generateBackupCodes();
  this.setState({ backupCodes, publicKeys});
}